### PR TITLE
Jaeger: Search traces

### LINKED ISSUE #24953: Searching jaeger traces by tags
**What would you like to be added**: Ability to search jaeger traces using tags

**Why is this needed**: I find that functionality very useful in default jaeger UI (https://github.com/jaegertracing/jaeger-ui), since sometimes I don't know exact trace ID, but I know how to find it using tags.

### LINKED ISSUE #27245: Support jaeger service and operation search with Grafana querystring
**What would you like to be added**: 
Support Jaeger service and operation search with Grafana querystring 

**Why is this needed**: 
We are doing dashboard hierarchy, it's convenient if we can click slow request url then jump to jaeger explore directly

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

This PR implements search query for Jaeger.

### TODO

- [x] Add images to docs
- [x] Update UI once Nico is ready with his feedback
- [x] Add tests

**Which issue(s) this PR fixes**:

Fixes #24953
Fixes #27245

### CODE REVIEW

**Review by achatterjee-grafana — COMMENTED** [review]
Added some review comments.

File: docs/sources/datasources/jaeger.md [hunk 1]
```diff
@@ -39,25 +39,24 @@ This is a configuration for the [trace to logs feature]({{< relref "../explore/t
...
+1. Limit - Limits the number of traces returned.
 
 ## Linking Trace ID from logs
 
 You can link to Jaeger trace from logs in Loki by configuring a derived field with internal link. See the [Derived fields]({{< relref "loki.md#derived-fields" >}}) section in the [Loki data source]({{< relref "loki.md" >}}) documentation for details.
 
 ## Configure the data source with provisioning
 
-You can set up the data source via configuration files with Grafana’s provisioning system. Refer to [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}) for information on various settings and how it works.
+You can set up the data source via configuration files with Grafana's provisioning system. Refer to [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}) for information on various settings and how it works.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/jaeger.md:59]:
```suggestion
You can set up the data source via configuration files with Grafana's provisioning system. Refer to [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}) for more information on configuring various settings.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Another suggestion.

File: docs/sources/datasources/jaeger.md [hunk 2]
```diff
@@ -39,25 +39,24 @@ This is a configuration for the [trace to logs feature]({{< relref "../explore/t
...
-
-Use the trace selector to pick particular trace from all traces logged in the time range you have selected in Explore. The trace selector has three levels of nesting:
-
-1. The service you are interested in.
-1. Particular operation is part of the selected service.
-1. Specific trace in which the selected operation occurred, represented by the root operation name and trace duration.
+1. Service - Lists services.
+1. Operation - Populated after selecting a service. Lists operations related to the selected service. Select [All] option to query all the operation.
+1. Tags - Use values in the [logfmt](https://brandur.org/logfmt) format. For example `error=true db.statement="select * from User"`.
+1. Min Duration - Setting this field should filter traces that has higher duration than the value. Possible values are `1.2s, 100ms, 500us`.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/jaeger.md:49]:
```suggestion
1. Min Duration - Filter all traces with a duration higher than the set value. Possible values are `1.2s, 100ms, 500us`.
```

File: docs/sources/datasources/jaeger.md [hunk 3]
```diff
@@ -39,25 +39,24 @@ This is a configuration for the [trace to logs feature]({{< relref "../explore/t
...
+1. Max Duration - Setting this field should filter traces that duration are not higher than the value. Possible values are `1.2s, 100ms, 500us`.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/jaeger.md:50]:
```suggestion
1. Max Duration - Filter all traces with a duration lower than the set value. Possible values are `1.2s, 100ms, 500us`.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added copy edit changes.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Comment by aocenas** [comment]
Code looks good but there are some behavior that I think we should try to fix:
1. The limit placeholder is not seen in full so it's not clear what it means, we should probably put longer description in some info tooltip.
Screenshot from 2021-05-04 15-21-29

2. When querying logs the order is not stable and I get random traces every time. This isn't happening in jaeger ui so I assume there is either some param in the API or they sort the traces by time after they get them.
3. There seems to be some weird behavior with min/max limits but some of that is also present in Jaeger. In Jaeger if I use both min + max it seems like max is ignored. In this PR it seems like max limit is always ignored.

**Comment by zoltanbedi** [comment]
> Code looks good but there are some behavior that I think we should try to fix:
> 
> 1. The limit placeholder is not seen in full so it's not clear what it means, we should probably put longer description in some info tooltip.
> 2. When querying logs the order is not stable and I get random traces every time. This isn't happening in jaeger ui so I assume there is either some param in the API or they sort the traces by time after they get them.
> 3. There seems to be some weird behavior with min/max limits but some of that is also present in Jaeger. In Jaeger if I use both min + max it seems like max is ignored. In this PR it seems like max limit is always ignored.

1. Added the placeholder text to the info tooltip instead.
2. It seems like that they sort by most recent traces so I changed it to do the same. Don't we have a field config related to sorting?
3. I couldn't reproduce this. I see the same behaviour as in the Jaeger UI.

**Comment by aocenas** [comment]
> I couldn't reproduce this. I see the same behaviour as in the Jaeger UI.

hmm yes you are right, but still this seems super weird, I mean this seems like a bug in the Jaeger API but I think this will create some issues for us as people will assume it's a bug in Grafana. @hemdrup @davkal what do you think about this? From perspective of Grafana user this seems broken. Should we allow only one of mix/max at the time? Or some disclaimer?

Another small nitpick:
Would probably change the placeholder to something like `foo, bar, baz` (or some more real tags) to give an example of the syntax (delimiter) as that is more helpful there imho.
Screenshot from 2021-05-05 15-06-40

**Comment by hemdrup** [comment]
> Should we allow only one of mix/max at the time? Or some disclaimer?


- @aocenas Great idea on the "tag" example
- @aocenas Not sure what you mean by "only one min/mac" Do you mean multiple allowing filters like? Show me logs between 2-3s + logs between 4-5s?

**Review by aocenas — APPROVED** [review]
We can check the min/max issue in later PR

**Pull request merged**